network/hostnetwork/packet/packet.go

```diff
@@ -123,24 +124,32 @@ func SerializePacket(q *Packet) ([]byte, error) {
 // DeserializePacket reads packet from io.Reader.
 func DeserializePacket(conn io.Reader) (*Packet, error) {
 
-	lengthBytes := make([]byte, 8)
-	if _, err := io.ReadFull(conn, lengthBytes); err != nil {
-		return nil, err
-	}
-	lengthReader := bytes.NewBuffer(lengthBytes)
-	length, err := binary.ReadUvarint(lengthReader)
+	r := bufio.NewReader(conn)
+	lengthBytes, err := r.Peek(8)
```

Comment:
The algorithm is incorrect because error `ErrBufferFull` is not handled. `ReadFull` blocks until the requested count of bytes are available in buffer, so it always returns the desired count of bytes or error. `Peek` can return error if there are only 6 bytes available at the current moment and the rest 2 bytes are being transferred and are not available in buffer yet.